src/lib.rs

```diff
@@ -996,7 +981,7 @@ impl Buffer {
             Box::into_raw(user_data) as *mut u8,
         );
 
-        future
+        future.await
```

Comment:
Doesn't this wrap `GpuFuture` into another future? It seems better to avoid that if we can (this is why we were returning`GpuFuture` directly here)

Reply:
While this is technically true, it won't add any measurable impact here. And in a few weeks, futures are getting a few optimizations through generator resume arguments, so it's possible that this could get entirely optimized away.